barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.hpp

```diff
@@ -591,6 +591,144 @@ template <typename Curve_> class IPA {
     {
         return reduce_verify_internal_recursive(opening_claim, transcript);
     }
+
```

Comment:
possibly could deduplicate some logic here since it shares a lot of similar things with the non full recursive verifier, but opted not to for now since the part that isn't shared is in the middle of the function and isn't a prefix/suffix.